holopy/scattering/tests/test_composite.py

```diff
@@ -0,0 +1,107 @@
+        self.assertEqual(set(spheres.parameters.keys()), expected_keys)
+
+    @attr("fast")
+    def test_manual_tie(self):
+        spheres = [Sphere(n=Uniform(0, 1), r=Uniform(0, 1)) for i in range(3)]
+        ties = {'r': ['0:r', '1:r', '2:r'], 'n': ['0:n', '1:n', '2:n']}
+        spheres = Spheres(spheres, ties=ties)
+        expected_keys = {'n', 'r'}
+        self.assertEqual(set(spheres.parameters.keys()), expected_keys)
+
+    @attr("fast")
+    def test_adding_tied_scatterer(self):
+        n = Uniform(0, 1)
+        r = Uniform(0, 1)
+        spheres = [Sphere(n=n, r=r, center=[i, i, i]) for i in range(2)]
+        spheres = Spheres(spheres)
+        spheres.add(Sphere(n=n, r=r, center=[2, 2, 2]))
+        expected_keys = {'n', 'r', '0:center.0', '0:center.1', '0:center.2',
+                                   '1:center.0', '1:center.1', '1:center.2',
+                                   '2:center.0', '2:center.1', '2:center.2'}
+        self.assertEqual(set(spheres.parameters.keys()), expected_keys)
+
+    @attr("fast")
+    def test_adding_newly_tied_scatterer(self):
+        n = Uniform(0, 1)
+        r = Uniform(0, 1)
+        spheres = [Sphere(n=n, r=r, center=[1, 1, 1])]
+        spheres = Spheres(spheres)
+        spheres.add(Sphere(n=n, r=r, center=[2, 2, 2]))
+        expected_keys = {'n', 'r', '0:center.0', '0:center.1', '0:center.2',
+                                   '1:center.0', '1:center.1', '1:center.2'}
+        self.assertEqual(set(spheres.parameters.keys()), expected_keys)
+
+    @attr("fast")
+    def test_adding_untied_scatterer(self):
+        n = Uniform(0, 1)
+        r = Uniform(0, 1)
+        spheres = [Sphere(n=n, r=r, center=[i, i, i]) for i in range(2)]
+        spheres = Spheres(spheres)
+        spheres.add(Sphere(n=1, r=1, center=[2, 2, 2]))
+        expected_keys = {'n', 'r', '0:center.0', '0:center.1', '0:center.2',
+                         '1:center.0', '1:center.1', '1:center.2', '2:n',
+                         '2:r', '2:center.0', '2:center.1', '2:center.2'}
+        self.assertEqual(set(spheres.parameters.keys()), expected_keys)
+
+    @attr("fast")
+    def test_from_parameters(self):
+        n = Uniform(0, 1)
+        r = Uniform(0, np.pi/2)
+        spheres = [Sphere(n=n, r=r, center=[i, i, i]) for i in range(3)]
+        spheres = Spheres(spheres)
+        parameters = spheres.parameters
+        self.assertEqual(spheres.guess, spheres.from_parameters(parameters))
+
+    @attr("medium")
+    def test_save_and_open(self):
+        n = Uniform(0, 1)
+        r = Uniform(0, np.pi/2)
+        spheres = [Sphere(n=n, r=r, center=[i, i, i]) for i in range(3)]
+        spheres = Spheres(spheres)
+        expected_keys = spheres.parameters.keys()
+        with tempfile.TemporaryDirectory() as tempdir:
+            hp.save(tempdir + '/test.out', spheres)
+            spheres = hp.load(tempdir + '/test.out')
+        self.assertEqual(spheres.parameters.keys(), expected_keys)
+
```

Comment:
I was looking through these tests to get an idea of what ```parameters``` and ```raw_parameters``` exactly returned, and I noticed that ```raw_parameters``` isn't tested anywhere. Can we add a unit test for ```raw_parameters```?

Also, if ```raw_parameters``` is not public, it might make sense to make that a little clearer (e.g. rename it ```_raw_parameters``` if outside class don't need to use it).